### PR TITLE
Add diagnostic for whitespace before number suffix

### LINKED ISSUE #1287: More useful error message for lengths with space
Currently
```
#box(stroke: 1 pt, [a #sym.zws]) 
```
yields the not very helpful error message
```
Error: expected comma
   ╭─[<user input>:8:15]
   │
 8 │ #box(stroke: 1 pt, [a #sym.zws])
   │                 
   │                  
───╯
```
It would be helpful if the error message could tell me something like "Did you mean 1pt?". For an end user, the error message is confusing, because there is a comma right after.

### PR DESCRIPTION
**Pull request by TomBinford**
Closes #1287

### CODE REVIEW

**Comment by TomBinford** [comment]
On the example from #1287 we now get these diagnostics:
```
error: whitespace between number and number suffix, did you mean 1pt?
  ┌─ \test.typ:1:13
  │
1 │ #box(stroke: 1 pt, [a #sym.zws])
  │              ^^^^

error: unexpected comma
  ┌─ \test.typ:1:17
  │
1 │ #box(stroke: 1 pt, [a #sym.zws])
  │                  ^
```
If we want to get rid of the second message badly enough we could try emitting tokens as if the first message was fixed, but that'd require changes to how the lexer reports errors.

File: crates/typst/src/syntax/lexer.rs [hunk 1]
```diff
@@ -591,6 +591,32 @@ impl Lexer<'_> {
...
+                excursion.eat_while(char::is_ascii_alphanumeric)
+            };
+            if matches!(
+                suffix_after_whitespace,
+                // Don't match "in"; we don't want to mistake the operator for the suffix.
+                "pt" | "mm" | "cm" | "deg" | "rad" | "em" | "fr" | "%"
+            ) {
+                self.s = excursion;
+                return self.error(eco_format!(
+                    "whitespace between number and number suffix, did you mean {}{}{}?",
```
Comment (istudyatuni) [inline crates/typst/src/syntax/lexer.rs:609]: Maybe write `number` and `suffix_after_whitespace` inline?

```suggestion
                    "whitespace between number and number suffix, did you mean {}{number}{suffix_after_whitespace}?",
```
But I'm not sure about common approach in typst

Comment (laurmaedje) [reply crates/typst/src/syntax/lexer.rs:609]: Where possible, I prefer writing it inline.

**Comment by laurmaedje** [comment]
While playing with this a bit, I noticed the following false positive:
```
#{
  let fr = 1
  fr = 2
}
```
We could catch this by only eating non-newlines in the excursion. That said, I wonder if there are other false positives we missed.

**Pull request closed without merging**